pkg/solidity-utils/test/foundry/utils/BaseTest.sol

```diff
@@ -65,16 +67,42 @@ abstract contract BaseTest is Test, GasSnapshot {
     // List of all ERC4626 tokens
     IERC4626[] internal erc4626Tokens;
 
+    bool private _initialized;
+
     // Default balance for accounts
-    uint256 internal defaultBalance = 1e9 * 1e18;
+    uint256 private _defaultAccountBalance = DEFAULT_BALANCE;
 
+    // ------------------------------ Initialization ------------------------------
     function setUp() public virtual {
         // Set timestamp only if testing locally
         if (block.chainid == 31337) {
             // Set block.timestamp to something better than 0
             vm.warp(START_TIMESTAMP);
         }
 
+        _initTokens();
+        _initAccounts();
+
+        // Must mock rates after giving wrapped tokens to users, but before creating pools and initializing buffers.
+        mockERC4626TokenRates();
+    }
+
+    function setDefaultAccountBalance(uint256 balance) internal {
+        if (_initialized) {
+            revert("Cannot change default account balance after initialization");
+        }
+        _defaultAccountBalance = balance;
+    }
+
+    function defaultAccountBalance() internal view returns (uint256) {
+        return _defaultAccountBalance;
```

Comment:
`require` that it be initialized? Could even lazy initialize it, if there's a reasonable default. Like if it's not initialized, set it to a global default and then set _initialized. If a particular test wants to use a different specific value, it can just be sure to call `setDefaultAccountBalance` before anything that references it.

Reply:
Or make a virtual `defaultAccountBalance()` that just returns the standard value, and if tests want something different, they can override it?